webpack/assets/javascripts/foreman_hosts.js

```diff
@@ -0,0 +1,39 @@
+export let pluginEditAttributes = {
+  architecture: [],
+  os: [],
+  medium: [],
+  image: []
+};
+
+export function getAttributesToPost(selectionType) {
+  let attrsToPost;
+
+  switch (selectionType) {
```

Comment:
this can maybe be a bit nicer with a dictionary object:
const defaultAttributes = { 'architecture': ['architecture_id', 'organization_id', 'location_id'] } etc...

return attrsToPost.concat(get(window, `tfm.hosts.pluginEditAttributes[${selectionType}]`) || []);

please try to use const instead of 'let', you code will be immutable and much safer